Add helper script for the build chain

n3f4s opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dev-osrose/osirose-new/94)
<!-- Reviewable:end -->

n3f4s committed:
Add helper for build-chain

n3f4s committed:
Add yad dependencie

n3f4s committed:
Debug helper script and add build script

RavenX8 commented:
Looks nice. Could you re-target the PR to the trunk branch?

RavenX8 reviewed (commented):
I would suggest making g++-6 the default version that this requires.

n3f4s commented:
I kept ``gcc-5`` because it was what was originally in the set-up script but IMHO ``gcc`` and ``g++`` should be used (without any version or path specification). As long as the compiler support ``c++14`` (and the other flags), the version won't cause problem (compiler bug aside). Specifying the path can be annoying for people that use link to modify the behaviour of the compiler (for example using ``/usr/local/bin/gcc`` as a link to ``ccache``).

There is some bug in the scripts that I have corrected but I have not pushed the fix yet (I will do it as soon as I can).

n3f4s committed:
export config in scripts/env.sh

menuconfig.sh write in scripts/env.sh
build.sh load env.sh if it exists or fallback to
default-env.sh

RavenX8 commented:
I agree with the gcc and g++ should be used instead of the gcc-<version> and g++-<version>. The next time I go through and fix up the CMake files I was going to convert them. The only reason why it currently uses the path is due to a issue on the Travis-CI where we can't overwrite set the default compiler.

L3nn0x commented on scripts/default-env.sh:
```diff
@@ -0,0 +1,7 @@
+export PROOT=/home/vincent/dev/osIROSE-new
```
You shouldn't hard code the paths. We don't all have a folder named vincent ;)

n3f4s replied:
oups, my bad, should have checked that ... (It's not my fault if bash can't read my mind and not expand the variable when it should but that's not what I wanted to do ! ^^  )

n3f4s committed:
use variables in absolute path for default env
